src/main/kotlin/de/terrestris/geoserver/wps/DistinctValues.kt

```diff
@@ -0,0 +1,193 @@
+import com.fasterxml.jackson.databind.node.JsonNodeFactory
+import net.sf.jsqlparser.JSQLParserException
+import net.sf.jsqlparser.expression.Function
+import net.sf.jsqlparser.expression.operators.conditional.AndExpression
+import net.sf.jsqlparser.expression.operators.relational.ExpressionList
+import net.sf.jsqlparser.parser.CCJSqlParserUtil
+import net.sf.jsqlparser.statement.select.PlainSelect
+import net.sf.jsqlparser.statement.select.Select
+import net.sf.jsqlparser.statement.select.SelectExpressionItem
+import net.sf.jsqlparser.statement.select.SelectItem
+import org.geoserver.config.GeoServer
+import org.geoserver.security.decorators.ReadOnlyDataStore
+import org.geoserver.wps.gs.GeoServerProcess
+import org.geoserver.wps.process.RawData
+import org.geoserver.wps.process.StringRawData
+import org.geotools.data.jdbc.FilterToSQLException
+import org.geotools.data.postgis.PostGISDialect
+import org.geotools.filter.text.cql2.CQLException
+import org.geotools.filter.text.ecql.ECQL
+import org.geotools.jdbc.JDBCDataStore
+import org.geotools.jdbc.VirtualTable
+import org.geotools.process.factory.DescribeParameter
+import org.geotools.process.factory.DescribeProcess
+import org.geotools.process.factory.DescribeResult
+import org.geotools.util.logging.Logging
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.SQLException
+import java.util.*
+import java.util.logging.Level
+import java.util.stream.Collectors
+
+@DescribeProcess(title = "distinctValues", description = "Gets the distinct values of a column.")
+class DistinctValues(private val geoServer: GeoServer) : GeoServerProcess {
+
+    private val objectMapper: ObjectMapper = ObjectMapper()
+
+    @DescribeResult(name = "result", description = "The distinct values json.", primary = true)
+    @Throws(
+        JsonProcessingException::class, SQLException::class
+    )
+    fun execute(
+        @DescribeParameter(
+            name = "layerName",
+            description = "The qualified name of the layer to retrieve the values from. The layer must be based on a " +
+                    "JDBC/postgres datastore and be based on a single table with a name equal to the layer name."
+        ) layerName: String,
+        @DescribeParameter(
+            name = "propertyName",
+            description = "The property name to retrieve the values of."
+        ) propertyName: String,
+        @DescribeParameter(name = "filter", description = "An optional CQL filter to apply.", min = 0) filter: String?,
+        @DescribeParameter(name = "viewParams", description = "Optional view params.", min = 0) viewParams: String?
+    ): RawData {
+        var conn: Connection? = null
+        var stmt: PreparedStatement? = null
+        var rs: ResultSet? = null
+        return try {
+            val factory = JsonNodeFactory(false)
+            val root = factory.arrayNode()
+            val tableName = layerName.split(":".toRegex()).toTypedArray()[1]
```

Comment:
```suggestion
            val tableName = layerName.split(":".toRegex())[1]
```
don't know if this is better

Reply:
Right, this is not necessary, I'm not sure why the conversion generated it like that